src/main/java/org/folio/template/dao/TemplateDaoImpl.java

```diff
@@ -30,61 +32,61 @@ public TemplateDaoImpl(Vertx vertx, String tenantId) {
 
   @Override
   public Future<List<Template>> getTemplates(String query, int offset, int limit) {
-    Future<Results<Template>> future = Future.future();
+    Promise<Results<Template>> promise = Promise.promise();
     try {
       String[] fieldList = {"*"};
       CQLWrapper cql = getCQL(query, limit, offset);
-      pgClient.get(TEMPLATES_TABLE, Template.class, fieldList, cql, true, false, future.completer());
+      pgClient.get(TEMPLATES_TABLE, Template.class, fieldList, cql, true, false, promise);
     } catch (Exception e) {
-      future.fail(e);
+      promise.fail(e);
```

Comment:
Can you add a unit test or an integration test for `promise.fail(e)`?

For example by adding

    TemplateDaoImpl(PostgresClient postgresClient) {
      pgClient = postgresClient;
    }

and testing with `new TemplateDaoImpl(null)`.